### ISSUE TITLE
[Feature Request] Overrideable settings in docfx.json 

### ISSUE THREAD

**Issue by CaseyHofland** [dotnet/docfx#8815]
**Is your feature request related to a problem? Please describe.**
I am creating a github action that does most of the heavy docfx lifting. However, I would like users to specify a file (e.g. `projectMetadata.json`) in which they can override docfx settings.

**Describe the solution you'd like**
Something like an "overwrite" in the `docfx.json` in which I can specify files e.g. `projectMetadata.json` to overwrite docfx settings.

**Describe alternatives you've considered**
Doing it manually using command line arguments inside the github action.

**Additional context**
Sorry guys it's me again, this is the last feature I need I swear then I've hit all my goals.

**Comment by yufeih** [comment]
The [globalMetadataFiles](https://github.com/dotnet/docfx/blob/main/src/Microsoft.DocAsCode.App/Config/BuildJsonConfig.cs#L50) config in `docfx.json` allows you to provide global metadata from a different file. It can override metadata not but other configs. Does that work for you?

**Comment by CaseyHofland** [comment]
I tried that and it seems like it _should_ work... but it doesn't.

I've added `"globalMetadataFiles": ["manual/projectMetadata.json"],` inside my build property and the file looks like this:
```
{
    "_appTitle": "Something Really Stupid",
    "_enableSearch": "false"
}
```

But search is still enabled and the app title is not Something Really Stupid.

Here is the docfx.json (though the syntax should be fine).
[docfx.txt](https://github.com/dotnet/docfx/files/11622927/docfx.txt)
